java/amazon-kinesis-producer/src/main/java/com/amazonaws/services/kinesis/producer/KinesisProducer.java

```diff
@@ -601,6 +626,20 @@ protected KinesisProducer(File inPipe, File outPipe) {
         
         return f;
     }
+
+    @AllArgsConstructor
+    private class FutureTimeoutRunnableTask implements Runnable {
+        private long id;
+
+        @Override
+        public void run() {
+            SettableFutureTracker futureTracker = getFutureTracker(id);
+            totalFutureTimeouts.getAndIncrement();
```

Comment:
My thought was instead of emitting on our side as a snapshot metric, we can emit it to be a cumulative metric and customers can themselves build the snapshot logic based on how often they want to snapshot and check the change in the total timeouts. 

Emitting from our side as a snapshot metric is a bit tricky since we don't know how often customers would want to check this metric and how would be decide which timeframe to consider at that time. So i feel this implementation is easier and clean for us as well as gives customers ability to build whatever metric math they want on their side.

Reply:
Let's remove exposing 'totalFutureTimeouts' for this iteration